library/junos_ping

```diff
@@ -0,0 +1,271 @@
+            interface=dict(required=False, default=None),
+            rapid=dict(required=False, default=True),
+            routing_instance=dict(required=False, default=None),
+            ttl=dict(required=False, default=None),
+            acceptable_packet_loss=dict(required=False, default=0),
+            count=dict(required=False, default='5')
+            ),
+        supports_check_mode=False
+    )
+
+    if not HAS_PYEZ:
+        module.fail_json(msg='junos-eznc >= 1.2.2 is required for this module')
+
+    results = {}
+    m_args = module.params
+
+    # Open connection to device
+    dev = Device(
+            m_args['host'],
+            user=m_args['user'],
+            passwd=m_args['passwd'],
+            port=m_args['port'],
+            gather_facts=False)
+
+    try:
+        dev.open()
+    except Exception as err:
+        msg = 'Unable to connect to {0}: {1}'.format(m_args['host'], str(err))
+        module.fail_json(msg=msg)
+        return
+
+    results['dest_ip'] = m_args['dest_ip']
+    results['count'] = m_args['count']
+    results['rapid'] = m_args['rapid']
+
+    results['changed'] = False
+
+    # Prepare parameters
+    ping_params = dict(
+            host=m_args['dest_ip'],
+            count=m_args['count'],
+            rapid=m_args['rapid'])
+
+    if m_args['source_ip'] is not None:
+        ping_params['source'] = m_args['source_ip']
+        results['source_ip'] = m_args['source_ip']
+
+    if m_args['routing_instance'] is not None:
+        ping_params['routing_instance'] = m_args['routing_instance']
+        results['routing_instance'] = m_args['routing_instance']
+
+    if m_args['interface'] is not None:
+        ping_params['interface'] = m_args['interface']
+        results['interface'] = m_args['interface']
+
+    if m_args['ttl'] is not None:
+        ping_params['ttl'] = str(m_args['ttl'])
+        results['ttl'] = str(m_args['ttl'])
+
+    ## Change default Timeout
+    timeout = int(m_args['timeout'])
+    if timeout > 0:
+        dev.timeout = timeout
+
+    try:
+
+        # Execute Ping
+        rpc_reply = dev.rpc.ping(**ping_params)
+
+    except Exception as err:
+        results['failed'] = True
+        results['msg'] = "unable to execute ping due to:{0}".format(err.message)
+        dev.close()
+        raise err
+
+    try:
+        # Try to Find probe summary
+        probe_summary = rpc_reply.find('probe-results-summary')
+        if probe_summary is None:
+            results['failed'] = True
+            results['msg'] = rpc_reply.findtext('rpc-error/error-message').strip()
+            dev.close()
+            module.exit_json(**results)
+
+        # Extract packet loss
+        packet_loss = rpc_reply.findtext("ping-results/probe-results-summary/packet-loss")
+        packets_sent = rpc_reply.findtext("ping-results/probe-results-summary/probes-sent")
+        responses_received = rpc_reply.findtext("ping-results/probe-results-summary/responses-received")
+
```

Comment:
Suggest adding the following lines so the individual return values are also easily accessible in the response, rather than having to parse them from results['msg']:

results['packet_loss'] = packet_loss
results['packets_sent'] = packets_sent
results['packets_received'] = packets_received